src/box/tuple_compare.cc

```diff
@@ -906,6 +906,46 @@ key_compare(const char *key_a, hint_t key_a_hint,
 	}
 }
 
+const char min_key = '\0';
+const char max_key = '\0';
+int
+key_compare_ext(const char *key_a, hint_t key_a_hint,
+		const char *key_b, hint_t key_b_hint, struct key_def *key_def)
+{
+	assert(key_a != NULL);
+	assert(key_b != NULL);
+
+	if (key_a == key_b && (key_a == &min_key || key_a == &max_key))
+		return 0;
+
+	if (key_a == &min_key)
+		return -1;
+	if (key_a == &max_key)
```

Comment:
I think that it is better to combine the conditions that give the same result under one if